src/Core/Platform/Infrastructure/AzureScaffolder.cs

```diff
@@ -0,0 +1,142 @@
+#nullable enable
+
+using Azure;
+using Azure.Data.Tables;
+using Azure.Storage.Blobs;
+using Azure.Storage.Blobs.Models;
+using Azure.Storage.Queues;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+
+namespace Bit.Core.Platform.Infrastructure;
+
+internal class BrokenDevelepmentEnvironmentException : Exception
+{
+    public BrokenDevelepmentEnvironmentException(string message)
+        : base (message)
+    {
+        
+    }
+
+    public BrokenDevelepmentEnvironmentException(string message, Exception innerException)
+        : base(message, innerException)
+    {
+
+    }
+}
+
+public class AzureScaffolder : IHostedService
+{
+    private static readonly IEnumerable<string> _containers =
+      [ "attachments", "sendfiles", "misc" ];
+
+    private static readonly IEnumerable<string> _queues =
+      [ "event", "notifications", "reference-events", "mail" ];
+
+    private static readonly IEnumerable<string> _tables =
+      [ "event", "metadata", "installationdevice"];
+
+    private static readonly BlobCorsRule _corsRule = new()
+    {
+        AllowedHeaders = "*",
+        ExposedHeaders = "*",
+        AllowedOrigins = "*",
+        MaxAgeInSeconds = 30,
+        AllowedMethods = "GET,PUT"
+    };
+
+    private static readonly EqualityComparer<BlobCorsRule> _corsComparer = EqualityComparer<BlobCorsRule>.Create((a, b) =>
+    {
+        if (a == null)
+        {
+            return b == null;
+        }
+
+        if (b == null)
+        {
+            return false;
+        }
+
+        return a.AllowedOrigins == b.AllowedOrigins
+            && a.AllowedMethods == b.AllowedMethods
+            && a.AllowedHeaders == b.AllowedHeaders
+            && a.ExposedHeaders == b.ExposedHeaders
+            && a.MaxAgeInSeconds == b.MaxAgeInSeconds;
+    });
+
+    private static readonly string _developmentUri = "UseDevelopmentStorage=true";
+    private readonly ILogger<AzureScaffolder> _logger;
+
+    public AzureScaffolder(ILogger<AzureScaffolder> logger)
+    {
+        _logger = logger;
+    }
+
+    public async Task StartAsync(CancellationToken cancellationToken)
```

Comment:
Any errors here, will stop the startup of the application. I could instead catch and not-rethrow errors to allow startup to continue but that will generally lead to the error showing up later in the application and likely not rethrowing a more descriptive error.